src/app.py

```diff
@@ -108,6 +110,65 @@ def contacts():
         app.log_exception(sys.exc_info())
         return Response("Error getting users", status=503)  # well, it's better than crashing
 
+@app.route('/ticket')
+def ticket():
+    return _fix_unicode(render_template('ticket.html.j2'))
+
+@app.route('/submitTicket', methods=['POST'])
+def submitTicket():
+    information = request.form.to_dict(flat=False)
+    # First, check that the request passed captcha
+    if 'g-recaptcha-response' not in information:
+        return "Hello Bot"
+    captcha_url = "https://www.google.com/recaptcha/api/siteverify"
+    captcha_data = {
+        'secret': app.config['GOOGLE_CAPTCHA_KEY'],
+        'response': information['g-recaptcha-response'][0]
+    }
+    captcha_response = requests.post(captcha_url, data = captcha_data)
+    json_response = captcha_response.json()
+    if json_response['success'] is False:
+        return "Failed to request Captcha information"
+
+    if json_response['score'] < 0.5:
+        return "Your captcha score was too low, not submitting ticket"
+
+    cc_emails = set()
+    for facility in global_data.get_topology().resources_by_facility.values():
+        for resource in facility:
+            if resource.name in information['resource-list']:
+                resource_tree = resource.get_tree(authorized = True)
+                contact_list = resource_tree['ContactLists']['ContactList']
+                for contact in contact_list:
+                    if contact["ContactType"] == "Administrative Contact":
+                        for inner_contact in contact['Contacts']['Contact']:
+                            cc_emails.add(inner_contact['Email'])
+
+    print(cc_emails)
+    ticket = {
+        'subject': "Test API ticket",
```

Comment:
We should change the default subject before this goes live.

Reply:
I believe this is fixed. Resolving